deepchecks/nlp/text_data.py

```diff
@@ -65,210 +62,116 @@ class TextData:
     properties : t.Optional[Union[pd.DataFrame, str]] , default: None
         The text properties for the samples. If None, no properties are set. If 'auto', the properties are calculated
         using the default properties. If a DataFrame is given, it must contain the properties for each sample as the raw
         text and identical index.
-    device : str, default: None
-        The device to use to calculate the text properties.
     """
 
-    _text: t.Sequence[str]
+    _text: np.ndarray
     _label: TTextLabel
-    index: t.Sequence[t.Any]
-    _task_type: t.Optional[TaskType]
-    _has_label: bool
-    _is_multilabel: bool = False
+    task_type: t.Optional[TaskType]
+    _tokenized_text: t.Optional[t.Sequence[t.Sequence[str]]] = None  # Outer sequence is np array
     name: t.Optional[str] = None
     _metadata: t.Optional[pd.DataFrame] = None
+    _metadata_types: t.Optional[t.Dict[str, str]] = None
     _properties: t.Optional[t.Union[pd.DataFrame, str]] = None
+    _properties_types: t.Optional[t.Dict[str, str]] = None
+    _original_text_index: t.Optional[t.Sequence[int]] = None  # Sequence is np array
 
     def __init__(
             self,
             raw_text: t.Optional[t.Sequence[str]] = None,
             tokenized_text: t.Optional[t.Sequence[t.Sequence[str]]] = None,
             label: t.Optional[TTextLabel] = None,
-            task_type: t.Optional[str] = None,
-            dataset_name: t.Optional[str] = None,
-            index: t.Optional[t.Sequence[t.Any]] = None,
+            task_type: str = 'other',
+            name: t.Optional[str] = None,
             metadata: t.Optional[pd.DataFrame] = None,
-            properties: t.Optional[t.Union[pd.DataFrame, str]] = None,
-            device: t.Optional[str] = None
+            properties: t.Optional[t.Union[pd.DataFrame]] = None,
     ):
         # Require explicitly setting task type if label is provided
         if task_type in [None, 'other']:
             if label is not None:
-                if isinstance(label, t.Sequence):
-                    if pd.notnull(label).any():
-                        raise DeepchecksValueError('task_type must be set when label is provided')
-                else:
-                    raise DeepchecksValueError('task_type must be set when label is provided')
-
+                raise DeepchecksValueError('task_type must be set when label is provided')
             self._task_type = TaskType.OTHER
         elif task_type == 'text_classification':
             self._task_type = TaskType.TEXT_CLASSIFICATION
         elif task_type == 'token_classification':
+            if tokenized_text is None:
+                raise DeepchecksValueError('tokenized_text must be provided for token_classification task type')
+            validate_tokenized_text(tokenized_text)
+            modified = [[str(token) for token in tokens_per_sample] for tokens_per_sample in tokenized_text]
+            self._tokenized_text = np.asarray(modified)
```

Comment:
Why do this modification for every tokenized_text?
Also, you already validate that its a sequence of sequence of strings... no?

Reply:
I validate its a sequence, here i change it to numpy array of strings

Reply:
yeah... but why?

Reply:
So its consistant and will help avoid problems down the line,
ex: labels[[0,1]] return different thing if it is a list or np array of pandas

Reply:
👍